src/json.hpp.re2c

```diff
@@ -8097,64 +8097,143 @@ class basic_json
+                }
 
-        @param[in,out] endptr  recieves a pointer to the first character after
-        the number
+                // if negative number, reverse sign
+                if (negative_sign)
+                {
+                    result *= -1;
+                }
+            }
 
-        @return the floating point number
-        */
-        double str_to_float_t(double* /* type */, char** endptr) const
-        {
-            return std::strtod(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
-        }
+            // read in explicit exponent and calculate real exponent.
+            if (*str == 'e' or *str == 'E')
+            {
+                cp = *++str;
+                bool negative_exp = cp == '-'; // read in exponent sign (+/-)
 
-        /*!
-        @brief parse floating point number
+                bool plus_or_minus = false;
+                if (cp == '-' or cp == '+')
+                {
+                     cp = *++str;
+                     plus_or_minus = true;
+                }
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+                int count = 0; // exponent calculation
+                if (! nl_isdigit(cp))
+                {
+                     if (plus_or_minus)
+                     {
+                         *--str;
+                     }
 
-        @param[in] type  the @ref number_float_t in use
+                     *--str;
+                     goto skip_loop;
+                }
 
-        @param[in,out] endptr  recieves a pointer to the first character after
-        the number
+                while (nl_isdigit(cp))
+                {
+                    constexpr int imax = std::numeric_limits<int>::max();
 
-        @return the floating point number
-        */
-        float str_to_float_t(float* /* type */, char** endptr) const
-        {
-            return std::strtof(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
+                    if ((imax - std::abs(exp) - (cp - '0')) / 10 > count)
+                    {
+                         count *= 10;
+                         count += cp - '0';
+                    }
+                    else
+                    {
+                        count = imax - exp;
+                        break;
+                    }
+
+                    cp = *++str;
+                }
+skip_loop:
+                exp += negative_exp ? -count : count;
+            }
+
+            // adjust number by powers of ten specified by format and exponent.
+            constexpr std::array<long double, 9> powerof10 = {
+                {1.e1L, 1.e2L, 1.e4L, 1.e8L, 1.e16L, 1.e32L, 1.e64L, 1.e128L, 1.e256L}
+            };
+
+            if (exp > std::numeric_limits<long double>::max_exponent10)
```

Comment:
Starting here, there a now comments in the code. Please briefly describe each conditional and loop.